rest_framework/fields.py

```diff
@@ -1521,12 +1521,15 @@ def __init__(self, *args, **kwargs):
         self.child.bind(field_name='', parent=self)
 
     def get_value(self, dictionary):
-        if self.field_name not in dictionary:
-            if getattr(self.root, 'partial', False):
-                return empty
         # We override the default field access in order to support
         # lists in HTML forms.
         if html.is_html_input(dictionary):
+            if getattr(self.root, 'partial', False):
+                if self.field_name not in dictionary:
+                    regexp = re.compile(html.HTML_LIST_REGEXP_FORMAT % re.escape(self.field_name))
+                    if not any(regexp.match(key) for key in dictionary.keys()):
+                        return empty
+
```

Comment:
I'm not comfortable moving the `empty` check entirely inside `is_html_input` - seems like we might be missing some failure cases that introduces that aren't covered by our test cases.

Any way around we could arrange this PR that has a smaller change footprint?